Grbl_Esp32/Custom/CoreXY.cpp

```diff
@@ -90,9 +90,25 @@ bool user_defined_homing(uint8_t cycle_mask) {
     pl_data->motion.systemMotion   = 1;
     pl_data->motion.noFeedOverride = 1;
 
-    for (int cycle = 0; cycle < 3; cycle++) {
-        AxisMask mask = homing_cycle[cycle]->get();
+    uint8_t cycle_count = (cycle_mask == 0) ? 3 : 1;  // if we have a cycle_mask, we are only going to do one axis
+
+    AxisMask mask = 0;
+    for (int cycle = 0; cycle < cycle_count; cycle++) {
+        // if we have a cycle_mask, do that. Otherwise get the cycle from the settings
+        if (cycle_mask == 0) {
+            mask = homing_cycle[cycle]->get();
+        } else {
+            mask = cycle_mask;
+        }
```

Comment:
To my eye this would look better as
```
mask = cycle_mask ? cycle_mask : homing_cycle[cycle]->get();
```
Perhaps my fondness for the ternary is an affectation - but I do like the way it captures the idea that it all boils down to setting the value of one variable.  I also like the fact that it shrinks the code vertically, making it easier to see the entire structure of loops.